lib/macho/macho_file.rb

```diff
@@ -382,10 +410,16 @@ def change_rpath(old_path, new_path)
     #  file.add_rpath("/usr/lib")
     #  file.rpaths # => ["/lib", "/usr/lib"]
     # @param path [String] the new runtime path
+    # @param options [Hash]
+    # @option options [Boolean] :strict (true) whether or not to fail loudly
+    #  with an exception if the change cannot be performed
     # @return [void]
     # @raise [MachO::RpathExistsError] if the runtime path already exists
-    def add_rpath(path)
-      raise RpathExistsError.new(path) if rpaths.include?(path)
+    def add_rpath(path, options = {})
+      if rpaths.include?(path)
+        return unless options.fetch(:strict, true)
+        raise RpathExistsError.new(path)
+      end
```

Comment:
This change needs to be reverted. The modification methods in `MachOFile` need to always raise an exception, independently of `options` (because `options` currently only matters if multiple Mach-O slices are involved).

(The added `options` argument needs to stay of course.)